agent/dockerclient/sdkclientfactory/sdkclientfactory.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
```

Comment:
why is this mostly copied? Is the plan to get rid of the other client factory before merging this?

Reply:
The original idea was to have two clients living side by side and migrate the API calls one-by-one over until all of them were under new sdk client. Each client needs its own clientfactory as the clients generated implement different interfaces. Then, once all of the API calls are migrated, clientfactory could be removed.